packages/medusa/src/api-v2/admin/regions/route.ts

```diff
@@ -1,20 +1,47 @@
+import { createRegionsWorkflow } from "@medusajs/core-flows"
+import { CreateRegionDTO } from "@medusajs/types"
 import { remoteQueryObjectFromString } from "@medusajs/utils"
 import { MedusaRequest, MedusaResponse } from "../../../types/routing"
-import { defaultAdminRegionFields } from "./query-config"
 
 export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
   const remoteQuery = req.scope.resolve("remoteQuery")
 
-  const variables = { filters: req.filterableFields }
-
   const queryObject = remoteQueryObjectFromString({
     entryPoint: "region",
-    variables,
-    fields: defaultAdminRegionFields,
+    variables: {
+      filters: req.filterableFields,
+      order: req.listConfig.order,
+      skip: req.listConfig.skip,
+      take: req.listConfig.take,
+    },
+    fields: req.listConfig.select as string[],
+  })
+
+  const { rows: regions, metadata } = await remoteQuery(queryObject)
+
+  res.json({
+    regions,
+    count: metadata.count,
+    offset: metadata.skip,
+    limit: metadata.take,
+  })
+}
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  const input = [
+    {
+      ...(req.validatedBody as CreateRegionDTO),
+    },
+  ]
+
+  const { result, errors } = await createRegionsWorkflow(req.scope).run({
+    input: { regionsData: input },
+    throwOnError: false,
```

Comment:
I think having too many configuration options makes the code a bit more verbose, and it's better to just do it either through exceptions, or through error object (I prefer the second one, but either is fine). WDYT?

Reply:
why don't we let the workflow throw the error, that a good question I don't remember